pkg/network/driver_interface.go

```diff
@@ -418,3 +568,29 @@ func newDDAPIFilter(direction, layer C.uint64_t, ifaceIndex int, isIPV4 bool) C.
 
 	return fd
 }
+
+// prepare N read buffers
+// and return the IoCompletionPort that will be used to coordinate reads.
+// danger: even though all reads will reference the returned iocp, buffers must be in-scope as long
+// as reads are happening. Otherwise, the memory the kernel is writing to will be written to memory reclaimed
+// by the GC
+func prepareCompletionBuffers(h windows.Handle, count int) (iocp windows.Handle, buffers []*_readbuffer, err error) {
+	iocp, err = windows.CreateIoCompletionPort(h, windows.Handle(0), 0, 0)
+	if err != nil {
+		return windows.Handle(0), nil, errors.Wrap(err, "error creating IO completion port")
+	}
+
+	sizeOfReadBuffer := unsafe.Sizeof(_readbuffer{})
+	buffers = make([]*_readbuffer, count)
+	for i := 0; i < count; i++ {
+		buffers[i] = (*_readbuffer)(C.malloc(C.ulonglong(sizeOfReadBuffer)))
```

Comment:
I'm concerned Go may think it owns this memory since `_readbuffer` is a Go type and has a byte array. It may be safer to have `data` be an `unsafe.Pointer`, use a constant size to the `C.malloc` call, and cast to `[1280]byte` when necessary.